includes/Core/Authentication/Google_Proxy.php

```diff
@@ -95,6 +96,76 @@ public function get_site_fields() {
 		);
 	}
 
+	/**
+	 * Fetch site fields
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Credentials $credentials Credentials instance.
+	 * @return array|WP_Error The response as an associative array or WP_Error on failure.
+	 */
+	public function fetch_site_fields( Credentials $credentials ) {
+		if ( ! $credentials->has() ) {
+			return new WP_Error( 'oauth_credentials_not_exist' );
+		}
+
+		$creds = $credentials->get();
+
+		$request_args = array(
+			'body' => array(
+				'site_id'     => $creds['oauth2_client_id'],
+				'site_secret' => $creds['oauth2_client_secret'],
+			),
+		);
+
+		$response = wp_remote_post( $this->url( self::OAUTH2_SITE_URI ), $request_args );
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		if ( wp_remote_retrieve_response_code( $response ) !== 200 ) {
+			return new WP_Error( 'non_200_response_code' );
+		}
```

Comment:
This is somewhat unprecise, as the `WP_Error` returned would not provide information on the real error here - which may be okay as a last resort, but we should try to use more specific errors. For example, below you run the check whether the response contains an `error` key, which is possible, and in that case the response status code would not be 200. So we'd want to in such a case return the code in `error` rather than a less precise error that the response didn't come back with a 200.

I think we should just remove this clause completely. If the entire request fails, that would be caught by the `WP_Error` from `wp_remote_post`, and if the request comes back with an error response, we should be okay relying on the `error` key in the response.

Reply:
Great, updated and pushed in the most recent push.